### ISSUE TITLE
pedantic issues prevent compiling!

### ISSUE THREAD

**Issue by rhl-** [emilk/loguru#9]
.../code/project/include/log/loguru.hpp:1168:69: error: use of non-standard escape character '\e' [-Werror,-Wpedantic]
        const char* terminal_black()      { return s_terminal_has_color ? "\e[30m" : ""; }

I can't compile with loguru because I get a bazillion errors like these in pedantic mode in clang!

What do I do!?

**Comment by emilk** [comment]
Loguru does some non-standard things, like using terminal colors. I could, and maybe should, disable the pedantic warning within loguru.hpp (`#pragma gcc diagnostic ignore "-Wpedantic"`) - I'll think about that a bit.

In the mean time you can either do the above manually:

``` C++
#pragma gcc diagnostic push
#pragma gcc diagnostic ignore "-Wpedantic"
#define LOGURU_IMPLEMENTATION 1
#include <loguru.hpp>
#pragma gcc diagnostic pop
```

But a much cleaner solution is to configure your build system to use `-isystem` instead of `-I` to specify the `loguru.hpp` path (and any third party libraris). This mutes *all* warnings generated by the library. See https://gcc.gnu.org/onlinedocs/cpp/System-Headers.html for more info.

**Comment by rhl-** [comment]
Ok, just to be clear, this is happening with clang

**Comment by emilk** [comment]
I've fixed this in commit 3d0c47ebad64feb9a4fd495f0c26ceba2686c292. `loguru.hpp` now declares itself as a `system_header` muting all warnings within it.

Note that when using printf-style functions (`LOG_F`, `CHECK_F` and the like) you will still have to mute a specific warning, namely `gnu-zero-variadic-macro-arguments`. Since this is a problem outside of `loguru.hpp` you have three choices:

* Pass `-Wno-gnu-zero-variadic-macro-arguments` to clang
* Use the stream logging stuff instead of format logging (`LOG_S(INFO) << "Foo"` instead of `LOG_F(INFO, "Foo")`)
* Don't use `-pedantic`

Please let me know if this closes the issue for you!
